libcloud/compute/drivers/dimensiondata.py

```diff
@@ -3790,6 +3805,303 @@ def ex_change_nic_network_adapter(self, nic_id, network_adapter_name):
+                          'privateIpv4').text = ex_primary_nic_private_ipv4
+
+        if ex_primary_nic_vlan is not None:
+            vlan_id = self._vlan_to_vlan_id(ex_primary_nic_vlan)
+            ET.SubElement(pri_nic, 'vlanId').text = vlan_id
+
+        if ex_primary_nic_network_adapter is not None:
+            ET.SubElement(pri_nic,
+                          "networkAdapter").text = \
+                ex_primary_nic_network_adapter
+
+        if isinstance(ex_additional_nics, (list, tuple)):
+            for nic in ex_additional_nics:
+                additional_nic = ET.SubElement(network_elm,
+                                               'additionalNic')
+
+                if (nic.private_ip_v4 is None and
+                        nic.vlan is None):
+                    raise ValueError("Either a vlan or private_ip_v4 "
+                                     "must be specified for each "
+                                     "additional nic.")
+
+                if (nic.private_ip_v4 is not None and
+                        nic.vlan is not None):
+                    raise ValueError("Either a vlan or private_ip_v4 "
+                                     "must be specified for each "
+                                     "additional nic. Not both.")
+
+                if nic.private_ip_v4 is not None:
+                    ET.SubElement(additional_nic,
+                                  'privateIpv4').text = nic.private_ip_v4
+
+                if nic.vlan is not None:
+                    vlan_id = self._vlan_to_vlan_id(nic.vlan)
+                    ET.SubElement(additional_nic, 'vlanId').text = vlan_id
+
+                if nic.network_adapter_name is not None:
+                    ET.SubElement(additional_nic,
+                                  "networkAdapter").text = \
+                        nic.network_adapter_name
+        elif ex_additional_nics is not None:
+            raise TypeError(
+                "ex_additional_NICs must be None or tuple/list")
+
+        if isinstance(ex_disks, (list, tuple)):
+            for disk in ex_disks:
+                disk_elm = ET.SubElement(server_uncustomized_elm, 'disk')
+                disk_elm.set('scsiId', disk.scsi_id)
+                disk_elm.set('speed', disk.speed)
+        elif ex_disks is not None:
+            raise TypeError("ex_disks must be None or tuple/list")
+
+        # tagid and tagname value pair should not co-exists
+        if ex_tagid_value_pairs is not None and ex_tagname_value_pairs is \
+                not None:
+            raise ValueError("ex_tagid_value_pairs and ex_tagname_value_pairs"
+                             "is mutually exclusive. Use one or the other.")
+
+        # Tag by ID
+        if ex_tagid_value_pairs is not None:
+            if not isinstance(ex_tagid_value_pairs, dict):
+                raise ValueError(
+                    "ex_tagid_value_pairs must be a dictionary."
+                )
+
+            if sys.version_info[0] < 3:
```

Comment:
http://python-future.org/compatible_idioms.html#iterating-through-dict-keys-values-items in future, this is the correct way to do it in Py2/3